src/Cache/BlockEsiCache.php

```diff
@@ -1,177 +0,0 @@
-<?php
-
-declare(strict_types=1);
-
-/*
- * This file is part of the Sonata Project package.
- *
- * (c) Thomas Rabaix <thomas.rabaix@sonata-project.org>
- *
- * For the full copyright and license information, please view the LICENSE
- * file that was distributed with this source code.
- */
-
-namespace Sonata\PageBundle\Cache;
-
-use Sonata\BlockBundle\Block\BlockContextManagerInterface;
-use Sonata\BlockBundle\Block\BlockRendererInterface;
-use Sonata\Cache\CacheElement;
-use Sonata\Cache\CacheElementInterface;
-use Sonata\Cache\Invalidation\Recorder;
-use Sonata\CacheBundle\Adapter\VarnishCache;
-use Sonata\PageBundle\CmsManager\CmsManagerInterface;
-use Symfony\Component\HttpFoundation\Request;
-use Symfony\Component\HttpFoundation\Response;
-use Symfony\Component\HttpKernel\Controller\ArgumentResolverInterface;
-use Symfony\Component\HttpKernel\Controller\ControllerResolverInterface;
-use Symfony\Component\HttpKernel\Exception\AccessDeniedHttpException;
-use Symfony\Component\HttpKernel\Exception\NotFoundHttpException;
-use Symfony\Component\Routing\Generator\UrlGeneratorInterface;
-use Symfony\Component\Routing\RouterInterface;
-
-/**
- * Cache block through an esi statement.
- *
- * @author Thomas Rabaix <thomas.rabaix@sonata-project.org>
- *
- * @final since sonata-project/page-bundle 3.26
- */
-class BlockEsiCache extends VarnishCache
```

Comment:
We need to add a deprecation on 3.x for all the classes removed.

`@deprecated` on the class, possibly trigger a deprecation in the constructor, and add an UPGRADE note.